v1/config.go

```diff
@@ -95,3 +95,14 @@ func ParseConfigFile(r io.ReadCloser) (*ConfigFile, error) {
 	}
 	return &cf, nil
 }
+
+func DeepCopyConfigFile(i Image) (*ConfigFile, error) {
```

Comment:
This raises an interesting problem with the mutability of the `ConfigFile`s and `Manifest`s returned by our interface.  A couple thoughts:
1. We should think about that more holistically (perhaps as a standalone PR),
1. We should simply add `.DeepCopy()` methods to `Manifest` and `ConfigFile`

Reply:
We decided to utilize k8s https://github.com/kubernetes/gengo/tree/master/examples/deepcopy-gen for the v1 package